File: Backend/notification-service/src/main/java/iuh/fit/notification_service/services/NotificationService.java
package iuh.fit.notification_service.services;

import iuh.fit.common_service.dtos.response.base.PageResponse;
import iuh.fit.notification_service.dtos.response.NotificationResponse;

public interface NotificationService {
    PageResponse<NotificationResponse> listMyNotifications(String identityUserId, int page, int limit);

    long countMyUnread(String identityUserId);

    void markAsRead(String identityUserId, Long notificationId);

    void markAllAsRead(String identityUserId);
}

File: Mobile/src/services/notification.service.ts
import axiosClient from '@/configurations/axios.config';
import { API_PREFIXES } from '@/constants/api-prefixes';
import type { PageResponse, ResponseSuccess } from '@/types/api-response';
import type { NotificationItem } from '@/types/notification';

const NOTIFICATION_PREFIX = API_PREFIXES.notifications;

export const notificationService = {
  listMyNotifications: async (page = 1, limit = 20) => {
    const { data } = await axiosClient.get<ResponseSuccess<PageResponse<NotificationItem>>>(
      `${NOTIFICATION_PREFIX}`,
      { params: { page, limit } }
    );
    return data;
  },

  getUnreadCount: async () => {
    const { data } = await axiosClient.get<ResponseSuccess<number>>(`${NOTIFICATION_PREFIX}/unread-count`);
    return data;
  },

  markAsRead: async (notificationId: number) => {
    const { data } = await axiosClient.patch<ResponseSuccess<void>>(
      `${NOTIFICATION_PREFIX}/${encodeURIComponent(String(notificationId))}/read`
    );
    return data;
  },

  markAllAsRead: async () => {
    const { data } = await axiosClient.patch<ResponseSuccess<void>>(`${NOTIFICATION_PREFIX}/read-all`);
    return data;
  },
};
